### PR TITLE
Apply on the ActivityCard component the correct site time zone

### PR DESCRIPTION
**Pull request by cleacos**
#### Changes proposed in this Pull Request

pass the props `ActivityCard` component will need to apply the correct site time zone. We pass the site TZ and the GMT offset to `ActivityCard` to apply it to the activity time.

Also, we refactor the props `siteTimezone` and `siteGmtOffset` on `BackupsPage` component to  `timezone` and `gmtOffset`to facilitate the use of the destructuring assignments.

#### Testing instructions

- Apply the changes
- Find a date with a backup with real-time changes (to see the Activity Card component in action)
- Check that nothing has changed vs master

### CODE REVIEW

File: client/landing/jetpack-cloud/components/activity-card/index.jsx [hunk 1]
```diff
@@ -19,13 +20,18 @@ import './style.scss';
 
 class ActivityCard extends Component {
 	render() {
-		const { activity, moment, allowRestore } = this.props;
+		const { activity, allowRestore, timezone, gmtOffset } = this.props;
+
+		const backupTimeDisplay = applySiteOffset( activity.activityTs, {
+			timezone,
+			gmtOffset,
+		} ).format( 'H:mm' );
```
Comment (mattgawarecki) [inline client/landing/jetpack-cloud/components/activity-card/index.jsx:28]: I noticed in the previous commit we were returning a string formatted with `'LT'`, which seems like it's locale-aware. Do we want to continue that here, or is `'H:mm'` an intentional choice?

```suggestion
		} ).format( 'LT' );
```

Comment (cleacos) [reply client/landing/jetpack-cloud/components/activity-card/index.jsx:28]: Hey @mattgawarecki, thank you to review it :) Yes, it's intentional, a few days ago I asked Yvonne if at the end the time format would be 0-23, and she confirmed it. Yes, I know that the design doesn't match

**Comment by cleacos** [comment]
@cbauerman Yes, it's expected in this PR. The BackupDelta component is updated in this PR https://github.com/Automattic/wp-calypso/pull/40752 and now the list of ActivityCard is rendered on the BackupsPage. I've just added these props to the component on #40752 to be ready when we merge both.

**Comment by cbauerman** [comment]
> Yes, it's expected in this PR

then it should _really_ be in the testing instructions! 🙂

> * Find a date with a backup with real-time changes (to see the Activity Card component in action)
> * Check if the hour of each activity matches with the site time zone. You can compare it on your WordPress Activity Log section.

Are not true for this PR!

File: client/landing/jetpack-cloud/components/activity-card/index.jsx (hunk 1 above)
Comment (cleacos) [reply client/landing/jetpack-cloud/components/activity-card/index.jsx:28]: @mattgawarecki, in the end, we are going to use the locale format LT, so I'm going to apply the change.

**Pull request merged**